lib/valueflow.cpp

```diff
@@ -5970,6 +5970,39 @@ struct ConditionHandler {
             return findPath(true_values) | findPath(false_values);
         }
 
+        Token* getContextAndValues(Token* condTok,
+                                   std::list<ValueFlow::Value>& thenValues,
+                                   std::list<ValueFlow::Value>& elseValues,
+                                   bool known = false) const
+        {
+            const MathLib::bigint path = getPath();
+            const bool allowKnown = path == 0;
+            const bool allowImpossible = impossible && allowKnown;
+
+            bool inverted2 = inverted;
+            Token* ctx = skipNotAndCasts(condTok, &inverted2);
+            bool then = inverted ? !inverted2 : true;
```

Comment:
This could be written as `!inverted || !inverted2`. Seems like we don't have `readability-simplify-boolean-expr` enabled?